server.js

```diff
@@ -6030,6 +6030,91 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Uptime Robot status integration.
+camp.route(/^\/uptimerobot\/status\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var monitorApiKey = match[1];  // eg, u956-afus321g565fghr519
+  var format = match[2];
+  var badgeData = getBadgeData('status', data);
+  var options = {
+    method: 'POST',
+    json: true,
+    body: {
+      "api_key": monitorApiKey,
+      "format": "json"
+    },
+    uri: 'https://api.uptimerobot.com/v2/getMonitors'
+  };
+  request(options, function(err, res, json) {
+    if (err !== null || res.statusCode >= 500 || typeof json !== 'object') {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
```

Comment:
If an incorrect api key is entered for a monitor the following json is returned :

```json
{
	"stat": "fail",
	"error": {
		"type": "invalid_parameter",
		"parameter_name": "api_key",
		"passed_value": "u956-afus321g565fghr519",
		"message": "api_key not found."
	}
}
```

I can add this case if this is what you want with something like that :

```javascript
...
if (json.stat === 'fail') {
    badgeData.text[1] = 'unknown error';
    if (json.error) {
        badgeData.text[1] = json.error.message;
    }
    badgeData.colorscheme = 'lightgrey';
    sendBadge(format, badgeData);
    return;
}
var status = json.monitors[0].status;
if (status === 0) {
```